### ISSUE TITLE
TypeError: a bytes-like object is required, not 'str' on OSX via brew

### ISSUE THREAD

**Issue by firstgear** [rdiff-backup/rdiff-backup#320]
On Mac OSX 10.14.6, getting below error when using rdiff-backup even for basic sync of folder with one file. Also rdiff-backup -V shows DEV as you can see in below output.
```bash
#uname -a
Darwin systemname 18.7.0 Darwin Kernel Version 18.7.0: Mon Feb 10 21:08:45 PST 2020; root:xnu-4903.278.28~1/RELEASE_X86_64 x86_64

#brew info rdiff-backup
rdiff-backup: stable 2.0.0 (bottled)
Reverse differential backup tool, over a network or locally
https://rdiff-backup.net/
/usr/local/Cellar/rdiff-backup/2.0.0 (77 files, 1.1MB) *
  Poured from bottle on 2020-04-09 at 14:02:06
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/rdiff-backup.rb
==> Dependencies
Required: librsync ✔, python@3.8 ✔
==> Analytics
install: 142 (30 days), 195 (90 days), 492 (365 days)
install-on-request: 119 (30 days), 168 (90 days), 460 (365 days)
build-error: 0 (30 days)

#brew install rdiff-backup
Updating Homebrew...
Warning: rdiff-backup 2.0.0 is already installed and up-to-date
To reinstall 2.0.0, run `brew reinstall rdiff-backup`

#brew reinstall rdiff-backup
==> Reinstalling rdiff-backup
==> Downloading https://homebrew.bintray.com/bottles/rdiff-backup-2.0.0.mojave.bottle.tar.gz
Already downloaded: /Users/user/Library/Caches/Homebrew/downloads/826bb93067c0e28e86eca75b93b702f43662e1fe8e0a8b82fe6fc4fe827af76d--rdiff-backup-2.0.0.mojave.bottle.tar.gz
==> Pouring rdiff-backup-2.0.0.mojave.bottle.tar.gz
🍺  /usr/local/Cellar/rdiff-backup/2.0.0: 45 files, 721.8KB

#rdiff-backup -V
rdiff-backup DEV

#ls -al ./source
total 8
drwxrwxrwx   3 user  group   96 Apr  9 14:00 .
drwx------+ 15 user  group  480 Apr  9 14:01 ..
-rw-------   1 user  group    5 Apr  9 14:00 file1.txt

#rm -rf ./destination

#rdiff-backup ./source ./destination
Exception 'a bytes-like object is required, not 'str'' raised of class '<class 'TypeError'>':
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/Main.py", line 390, in error_check_Main
    Main(arglist)
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/Main.py", line 412, in Main
    take_action(rps)
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/Main.py", line 348, in take_action
    Backup(rps[0], rps[1])
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/Main.py", line 424, in Backup
    rpout.conn.fs_abilities.backup_set_globals(rpin, force)
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/fs_abilities.py", line 1044, in backup_set_globals
    dest_fsa = FSAbilities('destination').init_readwrite(Globals.rbdir)
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/fs_abilities.py", line 167, in init_readwrite
    self.set_resource_fork_readwrite(subdir)
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/fs_abilities.py", line 524, in set_resource_fork_readwrite
    fp_write.write(s)

Traceback (most recent call last):
  File "/usr/local/bin/rdiff-backup", line 32, in <module>
    rdiff_backup.Main.error_check_Main(sys.argv[1:])
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/Main.py", line 390, in error_check_Main
    Main(arglist)
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/Main.py", line 412, in Main
    take_action(rps)
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/Main.py", line 348, in take_action
    Backup(rps[0], rps[1])
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/Main.py", line 424, in Backup
    rpout.conn.fs_abilities.backup_set_globals(rpin, force)
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/fs_abilities.py", line 1044, in backup_set_globals
    dest_fsa = FSAbilities('destination').init_readwrite(Globals.rbdir)
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/fs_abilities.py", line 167, in init_readwrite
    self.set_resource_fork_readwrite(subdir)
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/fs_abilities.py", line 524, in set_resource_fork_readwrite
    fp_write.write(s)
TypeError: a bytes-like object is required, not 'str'
```

**Comment by ericzolf** [comment]
These a few problems at once:

1. nobody tested MacOS X as we developed 2.0.0, despite multiple queries, so sorry for this, but we can't test without Mac (and nobody offered us a Mac to test ourselves). See e.g. #82 and the reason why we removed MacOS from the officially supported platforms.
2. this said, we'll do our best to help you but brew is a packaging format, so I guess the packager also needs to be contacted (by you). Based on our experience with the version string, the DEV thing is generally a packaging issue.
3. I have no clue of brew but looking at the [rdiff-backup.rb](https://github.com/Homebrew/homebrew-core/blob/master/Formula/rdiff-backup.rb), I'd say that it might be missing setuptools_scm as build dependency (I'm also surprised to not see something like librsync-dev(el), but that might be OK).
4. regarding the actual problem byte vs. str, yep, if nobody tests, that's what happens... I can try to fix "blind" those issues as users discover them, but it'll take time.
5. can you please cut & paste here the output of `rdiff-backup --verbose 9`?

So, long story short, we could use a developer with a Mac willing to help, else we can only do our best step by step, with the support (and understanding) of users like you.

**Comment by ericzolf** [comment]
Are you able to check the code in PR #321 and tell me if there is more to fix?

**Comment by firstgear** [comment]
Sure thanks for you swift response! As asked in point 5, please find below output of verbosity 9.

```
#rdiff-backup --verbosity 9
2020-04-09 15:44:10.631322 +0200  <CLIENT-99983>  Using rdiff-backup version DEV
2020-04-09 15:44:10.632204 +0200  <CLIENT-99983>  	with cpython /usr/local/opt/python@3.8/bin/python3.8 version 3.8.2
2020-04-09 15:44:10.645478 +0200  <CLIENT-99983>  	on macOS-10.14.6-x86_64-i386-64bit, fs encoding utf-8
2020-04-09 15:44:10.645616 +0200  <CLIENT-99983>  Fatal Error: No arguments given
See the rdiff-backup manual page for more information.### 
```

**Comment by firstgear** [comment]
> Are you able to check the code in PR #321 and tell me if there is more to fix?

Checked in the fix and ran usual backup job, now getting following output.
```bash
Exception 'Can't mix strings and bytes in path components' raised of class '<class 'TypeError'>':
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/Main.py", line 390, in error_check_Main
    Main(arglist)
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/Main.py", line 412, in Main
    take_action(rps)
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/Main.py", line 364, in take_action
    RemoveOlderThan(rps[0])
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/Main.py", line 933, in RemoveOlderThan
    rootrp = require_root_set(rootrp, 0)
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/Main.py", line 905, in require_root_set
    Globals.rbdir.conn.fs_abilities.single_set_globals(
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/fs_abilities.py", line 1109, in single_set_globals
    ssg.set_chars_to_quote(Globals.rbdir)
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/fs_abilities.py", line 980, in set_chars_to_quote
    ctq_rp = rbdir.append(b"chars_to_quote")
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/rpath.py", line 1410, in append
    return self.__class__(self.conn, self.base, self.index + ext)
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/rpath.py", line 1082, in __init__
    self.setdata()
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/rpath.py", line 1110, in setdata
    self.conn.rpath.setdata_local(self)
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/rpath.py", line 1825, in setdata_local
    rpath.get_resource_fork()
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/rpath.py", line 1691, in get_resource_fork
    os.path.join(self.path, '..namedfork', 'rsrc'), 'rb')
  File "/usr/local/Cellar/python@3.8/3.8.2/Frameworks/Python.framework/Versions/3.8/lib/python3.8/posixpath.py", line 90, in join
    genericpath._check_arg_types('join', a, *p)
  File "/usr/local/Cellar/python@3.8/3.8.2/Frameworks/Python.framework/Versions/3.8/lib/python3.8/genericpath.py", line 155, in _check_arg_types
    raise TypeError("Can't mix strings and bytes in path components") from None

Traceback (most recent call last):
  File "/usr/local/bin/rdiff-backup", line 32, in <module>
    rdiff_backup.Main.error_check_Main(sys.argv[1:])
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/Main.py", line 390, in error_check_Main
    Main(arglist)
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/Main.py", line 412, in Main
    take_action(rps)
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/Main.py", line 364, in take_action
    RemoveOlderThan(rps[0])
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/Main.py", line 933, in RemoveOlderThan
    rootrp = require_root_set(rootrp, 0)
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/Main.py", line 905, in require_root_set
    Globals.rbdir.conn.fs_abilities.single_set_globals(
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/fs_abilities.py", line 1109, in single_set_globals
    ssg.set_chars_to_quote(Globals.rbdir)
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/fs_abilities.py", line 980, in set_chars_to_quote
    ctq_rp = rbdir.append(b"chars_to_quote")
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/rpath.py", line 1410, in append
    return self.__class__(self.conn, self.base, self.index + ext)
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/rpath.py", line 1082, in __init__
    self.setdata()
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/rpath.py", line 1110, in setdata
    self.conn.rpath.setdata_local(self)
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/rpath.py", line 1825, in setdata_local
    rpath.get_resource_fork()
  File "/usr/local/Cellar/rdiff-backup/2.0.0/libexec/rdiff_backup/rpath.py", line 1691, in get_resource_fork
    os.path.join(self.path, '..namedfork', 'rsrc'), 'rb')
  File "/usr/local/Cellar/python@3.8/3.8.2/Frameworks/Python.framework/Versions/3.8/lib/python3.8/posixpath.py", line 90, in join
    genericpath._check_arg_types('join', a, *p)
  File "/usr/local/Cellar/python@3.8/3.8.2/Frameworks/Python.framework/Versions/3.8/lib/python3.8/genericpath.py", line 155, in _check_arg_types
    raise TypeError("Can't mix strings and bytes in path components") from None
TypeError: Can't mix strings and bytes in path components
```

**Comment by ericzolf** [comment]
I pushed a few more fixes to the PR #321 if you'd like to test them.

**Comment by firstgear** [comment]
Yes, awesome, everything is working again. Love rdiff-backup ! Happy I don't have to revert to timemachine on OSX !
